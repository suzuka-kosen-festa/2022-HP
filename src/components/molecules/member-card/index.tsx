import type { FC } from "react";
import tw, { css } from "twin.macro";
import type { MemberCardProperties } from "./types/model";

const Position = tw.p`font-zen font-bold underline text-xs sm:text-2xl`;

const Name = tw.p`font-zen font-bold text-xs sm:text-2xl`;

const MemberCard: FC<MemberCardProperties> = ({ imagePath, name, position = "other" }) => (
  <figure>
    <img
      css={css`
        aspect-ratio: 3 / 2;
      `}
      src={imagePath}
      alt={name}
      width={450}
      height={300}
      loading="lazy"
    />
    <figcaption css={[tw`flex items-center justify-center`, position !== "other" ? tw`space-x-4 sm:space-x-8` : null]}>
      {position === "other" ? null : <Position>{position === "leader" ? "LEADER" : "SUB LEADER"}</Position>}
      <Name>{name}</Name>
    </figcaption>
  </figure>
);

export { MemberCard };
