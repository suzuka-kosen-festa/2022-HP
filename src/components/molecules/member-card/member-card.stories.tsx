import type { ComponentStoryObj, ComponentMeta } from "@storybook/react";
import { ComponentPropsWithoutRef } from "react";
import { MemberCard } from ".";

type T = typeof MemberCard;
type Story = ComponentStoryObj<T>;
type Meta = ComponentMeta<T>;

const args: ComponentPropsWithoutRef<T> = {
  imagePath: "/statics/members/rintaro_itokawa.webp",
  name: "Rintaro Itokawa",
};

export default {
  args,
  argTypes: {
    imagePath: {
      description: "Pass the path to photo",
      control: {
        type: "text",
      },
    },
    name: {
      description: "Name of member",
      control: {
        type: "text",
      },
    },
    position: {
      description: "Position in the members",
      control: {
        type: "text",
      },
    },
  },
  component: MemberCard,
} as Meta;

export const Default: Story = {};
export const LEADER: Story = {
  args: {
    imagePath: "/statics/members/rikuto_kuwahara.webp",
    name: "Rikuto Kuwahara",
    position: "leader",
  },
};
export const SUB: Story = {
  args: {
    imagePath: "/statics/members/shoma_kobayashi.webp",
    name: "Shoma Kobayashi",
    position: "sub",
  },
};
