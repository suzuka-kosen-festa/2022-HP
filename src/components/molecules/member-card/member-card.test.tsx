import { composeStories } from "@storybook/testing-react";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import * as stories from "./member-card.stories";

const { Default } = composeStories(stories);

const options = {
  name: "",
};

describe("(components) molecules/member", () => {
  test("to be molecules", () => {
    const { container } = render(<Default />);
    expect(container).toBeMolecule();
  });
  test("to be [role=figure]", () => {
    const { getByRole } = render(<Default />);
    expect(getByRole("figure", options));
  });
  test("take snap shot", () => {
    const { container } = render(<Default />);
    expect(container).toMatchSnapshot();
  });
});
