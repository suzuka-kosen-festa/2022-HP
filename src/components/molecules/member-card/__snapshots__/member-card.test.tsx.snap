// Vitest Snapshot v1

exports[`(components) molecules/member > take snap shot 1`] = `
<div>
  <figure>
    <img
      alt="Rintaro Itokawa"
      class="css-8lx7w1-MemberCard"
      height="300"
      loading="lazy"
      src="/statics/members/rintaro_itokawa.webp"
      width="450"
    />
    <figcaption
      class="css-1mlw4fg-MemberCard"
    >
      <p
        class="css-uqq0xe"
      >
        Rintaro Itokawa
      </p>
    </figcaption>
  </figure>
</div>
`;
